File: app/src/main/java/com/kacper/krakgo/screens/sign_in/register/RegisterContract.java
package com.kacper.krakgo.screens.sign_in.register;

/**
 * Created by kacper on 22/10/2017.
 */

public interface RegisterContract {
    interface Presenter{
        void signInWithEmail(String username, String email);
    }
    interface View{
        void onSignUpCompleted();

        void showError( Exception exception);
    }

}

File: app/src/main/java/com/kacper/krakgo/screens/sign_in/register/RegisterContract.kt
package com.kacper.krakgo.screens.sign_in.register

import com.kacper.krakgo.mvp.MvpPresenter
import com.kacper.krakgo.mvp.MvpView

object RegisterContract {
    interface Presenter : MvpPresenter<View> {
        fun signInWithEmail(username: String, email: String)
    }

    interface View : MvpView {
        fun onSignUpCompleted()
    }
}